UnityProject/Assets/Editor/LevelBuilder/GUI/LevelBuilderWindow.cs

```diff
@@ -74,15 +77,79 @@ public void OnGUI()
 	{
 		// Level buttons.
 		GUILayout.Label("Select a level below to work on:");
-		foreach (string level in AssetFetcher.GetLevels()) 
-			if (GUILayout.Button(new GUIContent(level, ""))) 
-				AssetFetcher.WorkOnLevel(level);
+
+		// Get all levels currently in the game.
+		IList<string> levelList = AssetFetcher.GetLevels();
+
+		// Popup() expects a string[] array, so convert it.
+		string[] levels = levelList.ToArray<string>();
+
+		// optionChosenByUser initialised to 0, so the dropdown will be blank. 
+		// After an option is chosen for the first time, we update this.
+
+		optionChosenByUser--;
+
+		optionChosenByUser = EditorGUILayout.Popup(optionChosenByUser, levels);
+
+		// Index is returned and levels start from 1 so incrementing it.
+		optionChosenByUser++;
+
+		if (GUILayout.Button (new GUIContent ("Work!")))
+			AssetFetcher.WorkOnLevel ("Level" + optionChosenByUser);
 		
-		// Only let the level designer operate if it's a level. 
+		// Only let the level designer operate if it's a level.
+		// ie. If the scene is called "Main" then it is not a level.
+		// See the file hierarchy to understand.
 		if (SceneManager.GetActiveScene().name.Equals("Main"))
 			return;
 
+		if (GUILayout.Button (new GUIContent ("Create a new level."))) {
+
+			// We have a bug where Unity doesn't recognise our dynamically 
+			// added objects through the level builder so we need to force
+			// the change to be saved.
+			EditorSceneManager.MarkAllScenesDirty();
+			EditorSceneManager.SaveOpenScenes ();
+
+			Scene newScene = SceneHandler.createScene();
+
+			// Unload the scenes so we can create a new one.
+			SceneHandler.unloadScenes();
+
+			// Calling countScenes() inside SceneHandler to find the new 
+			// filename to save our scene with. 
+			int currentNoOfLevels = SceneHandler.countScenes ();
+			currentNoOfLevels++;
+
+			// Setting up the camera.
+			GameObject cameraGameObject = new GameObject("Main Camera");
+			cameraGameObject.AddComponent<Camera>();
+			cameraGameObject.tag = "MainCamera";
+
+			LevelBuilderWindow.SetUpScene ();
+
+			// Setting up the light.
+			GameObject directionalLight = new GameObject ("Directional Light");
+			Light lightSettings = directionalLight.AddComponent<Light> ();
+			lightSettings.type = LightType.Directional;
+
+			// The color needs to be normalized to the 0-1 range.
+			lightSettings.color = new Color (1.0f, 0.956f, 0.839f, 1.0f);
+			lightSettings.intensity = 1.0f;
+			lightSettings.shadowBias = 0.05f;
+			lightSettings.shadowNormalBias = 0.4f;
+			lightSettings.shadowNearPlane = 0.2f;
+			directionalLight.transform.position = new Vector3 (0.0f, 3.0f, 0.0f);
+			directionalLight.transform.rotation = Quaternion.Euler(50.0f, -30.0f, 0.0f);
+
+			if (!EditorSceneManager.SaveScene (newScene, "Assets/Scenes/Levels/Level" + currentNoOfLevels + ".unity")) {
+				Debug.Log ("The savescene line failed!!!!!!!!!!!!!");
```

Comment:
NIT: Maybe we should use [LogWarning](https://docs.unity3d.com/ScriptReference/Debug.LogWarning.html) or LogError?

Reply:
Yes. That is definitely the plan. I didn't want to make this PR too large (for obvious reasons :)))) that's why I left some unfinished things that are suggested (or will be) in #41.

Reply:
Fair :+1: